xwiki-platform-core/xwiki-platform-ratings/xwiki-platform-ratings-ui/src/main/resources/XWiki/Ratings.xml

```diff
@@ -394,9 +394,10 @@ position: relative;
     <className>XWiki.UIExtensionClass</className>
     <guid>d2420ff0-a5b9-4fae-a798-fc623aac7a6b</guid>
     <property>
-      <content>{{include reference='XWiki.RatingsMacros'/}}
-
-{{velocity}}
+      <content>{{velocity}}
+#if (!$isGuest)
```

Comment:
ok, if preventing guests from voting is already implemented then perfect!

So yes we need to check rights but probably not just view access to the xclass but also write access to create the voting pages.